queries/slippage/subquery_batchwise_internal_transfers.sql

```diff
@@ -185,8 +201,15 @@ buffer_trades as (
     where (
             case 
                 -- in order to classify as buffer trade, the postive surplus must be in an allow_listed token
-                when (a.amount > 0 and b.amount < 0 and a.token in (Select * from allow_listed_tokens)) 
-                    or (b.amount > 0 and a.amount < 0 and b.token in (Select * from allow_listed_tokens))
+                when ((a.amount > 0 and b.amount < 0 and a.token in (Select * from allow_listed_tokens)) 
+                    or (b.amount > 0 and a.amount < 0 and b.token in (Select * from allow_listed_tokens)))
+                    and
+                        -- We know that settlements have internal buffer trades if 
+                        -- either no amm interations are made 
+                        -- or the solution must come from a internal_buffer_trader_solvers solver
+                        -- Hence, in order to classify as buffer trade, also the following constraint must hold:
+                        (a.solver_address in (select * from internal_buffer_trader_solvers) 
+                        or a.count_amm_interations = 0)
```

Comment:
We definitely do not need to count these.. We should have been excluding them right from the top of the file. 

Update whatever query we have to say `where dex_swaps > 0` globally.